ui/src/submissions/common/components/SubmitButton.jsx

```diff
@@ -1,22 +1,29 @@
-import React, { Component } from 'react';
+import React, { useEffect } from 'react';
 import PropTypes from 'prop-types';
 import { Button } from 'antd';
+import usePrevious from '../../../common/hooks/usePrevious';
 
-class SubmitButton extends Component {
-  render() {
-    const { isSubmitting, isValid, isValidating } = this.props;
+function SubmitButton({ isSubmitting, isValidating, isValid }) {
+  const previousIsSubmitting = usePrevious(isSubmitting);
 
-    return (
-      <Button
-        type="primary"
-        htmlType="submit"
-        loading={isSubmitting || isValidating}
-        disabled={!isValid}
-      >
-        Submit
-      </Button>
-    );
-  }
+  useEffect(
+    () => {
+      if (previousIsSubmitting && !isSubmitting && !isValid) {
```

Comment:
Perhaps extract this line, to a meaningful variable so that it's more readable:

Example:
`const hasTriedToSubmitInvalidForm = previousIsSubmitting && !isSubmitting && !isValid;`